### ISSUE TITLE
getters/setters in 1.0.0-alpha.6

### ISSUE THREAD

**Issue by C45tr0** [RobinBuschmann/sequelize-typescript#516]
I am trying to add a getter and setter for a property following the docs, but I am receiving an error. 

```
@AllowNull(false)
@ForeignKey(() => Bar)
@Column
public get barKey(): string {
    let val = this.getDataValue('barKey');
    if (val) {
        val = val.toLowerCase();
    }
    return val;
}

public set barKey(val) {
    this.setDataValue('barKey', val.toLowerCase());
}
```

When running tsc I receive the following errors:
```
src/models/foo.model.ts:53:33 - error TS2345: Argument of type '"barKey"' is not assignable to parameter of type '"id" | "createdAt" | "updatedAt" | "deletedAt" | "version" | ({ [P in keyof this]: P; } & { dataValues: never; id: never; createdAt: never; updatedAt: never; deletedAt: never; version: never; ... 29 more ...; toJSON: never; } & { ...; })[keyof this]'.

53     let val = this.getDataValue('barKey');
                                   ~~~~~~~~

src/models/foo.model.ts:61:23 - error TS2345: Argument of type '"barKey"' is not assignable to parameter of type '"id" | "createdAt" | "updatedAt" | "deletedAt" | "version" | ({ [P in keyof this]: P; } & { dataValues: never; id: never; createdAt: never; updatedAt: never; deletedAt: never; version: never; ... 29 more ...; toJSON: never; } & { ...; })[keyof this]'.

61     this.setDataValue('barKey', val.toLowerCase());
                         ~~~~~~~~

```
Any help would be appreciated.

**Comment by RobinBuschmann** [comment]
Hmmm 🤔... typescript doesn't seem to solve `this` properly. Don't know why. When setting `this` explicitly like so..
```ts
public get barKey(this: YourModel /* <-- */): string {
    let val = this.getDataValue('barKey');
    if (val) {
        val = val.toLowerCase();
    }
    return val;
}
```
.. it should work. But it would be better if it would work without setting `this`. Need to check this. Do you have any ideas?

**Comment by C45tr0** [comment]
That does indeed fix the issue. Only have to put it on the get and it also fixes the set. 

I do not have any ideas currently. I'll think about it.

**Comment by RobinBuschmann** [comment]
This is fixed with #546 and released in `sequelize-typescript@1.0.0-alpha.8`
